sumologic-app-utils/sumo_app_utils.yaml

```diff
@@ -1,48 +1,93 @@
 AWSTemplateFormatVersion: '2010-09-09'
 Transform: 'AWS::Serverless-2016-10-31'
 Description: >
-    This solution consists of a lambda function which which gets triggered when CF stack is deployed. This is used for creating sumologic resources like collector, source and app folders.
+  This solution consists of a lambda function which which gets triggered when CF stack is deployed. This is used for creating sumologic resources like collector, source and app folders.
 
 Globals:
-    Function:
-        Timeout: 300
+  Function:
+    Timeout: 300
+
 Metadata:
   AWS::ServerlessRepo::Application:
     Author: Sumo Logic
     Description: This solution consists of a lambda function which which gets triggered when CF stack is deployed. This is used for creating sumologic resources like collector, source and app folders.
     HomePageUrl: https://github.com/SumoLogic/sumologic-aws-lambda
     Labels:
-    - lambda
-    - sumologic
-    - serverless
+      - lambda
+      - sumologic
+      - serverless
     Name: sumologic-app-utils
-    SemanticVersion: 1.0.24
+    SemanticVersion: 2.0.0
     SourceCodeUrl: https://github.com/SumoLogic/sumologic-aws-lambda/tree/master/sumologic-app-utils
     SpdxLicenseId: Apache-2.0
+    ReadmeUrl: ./README.md
+
+Parameters:
+
+  RoleArn:
+    Type: String
+    Description: "Attach any existing Role ARN to Lambda Function. If no Existing Role ARN is provided, a role will be created and attached to the lambda function."
+    Default: ""
+
+Conditions:
+  role_arn: !Not [!Equals [!Ref RoleArn, '']]
+  No_role_arn: !Equals [!Ref RoleArn, '']
 
 Resources:
+  SumoAppUtilsFunctionWithRole:
+    Type: 'AWS::Serverless::Function'
+    Condition: role_arn
```

Comment:
I thought of the same. But kept it as it is because there can be may action that will be required based on the usage. People can add required action and resources in separate roles and then use the SAM app with role ARN gives better control over action and resources in IAM roles.

Reply:
customers can't change the role nor edit the SAM app can you give example of what you mean by change in action by usage?

Reply:
Lets say someone want to use the SAM app just for Collector and Source creation, then there is no need for providing tagging actions in IAM role. So, if we create role in the SAM templates, then it should always have all the action associated which Sumo app utils required thus not giving better control over actions.

But if we take role arn as input, then some one can create a IAM role with there own set of actions and then can pass it to SAM app to be used with lambda function.

Reply:
I don't think you got my point I am not against creating a role. There are two ways of creating a role one is directly attaching to the lambda other(the way you have done) is creating IAM::Role as a resource separately (provided it's supported by SAM). Here role creation and lambda creation will be created separately.

Reply:
I think you not getting my point. I am saying that it will be better if do not create an IAM role here with all actions that are needed for SumoAppUtils to preform operation. We can keep it to the person, who is using the SAM app and then they can provide a Role ARN that will be just attached to the lambda. If they do not provide a role ARN, we are still creating one where lambda internally creates one role.